src/frontend.c

```diff
@@ -262,9 +266,36 @@ void destroy_win(WINDOW *local_win) {
   delwin(local_win);
 }
 
-int print_status(char *str, WINDOW *window) {
+/* va_list version of mvwprintwf
+ */
+int vmvwprintwf(WINDOW *window, int y, int x, char *format, va_list argp) {
+  char buffer[64];
+  vsnprintf(buffer, sizeof(buffer) / sizeof(char), format, argp);
+  return mvwprintw(window, y, x, buffer);
+}
+
+/* A variation of mvwprintw that accepts format strings like printf
```

Comment:
Oops, looks like this is what `ncurses` does already, I just assumed it took a single string like `print_status`. I should have read the docs!